### PR TITLE
feat: configured quartz scheduler to fire nightly case stayed jobs (CMC-482)

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-482

### Change description ###

Added quartz scheduler configuration classes and db table sql

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by swalker125** [comment]
Screen Shot 2020-07-21 at 16 40 41
Screen Shot 2020-07-21 at 16 40 52

**Review by dharmendrak — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/unspec/config/scheduler/JobFactory.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.scheduling.quartz.SpringBeanJobFactory;
+import org.springframework.stereotype.Component;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Enables DI in classes bootstrapped by quartz.
+ */
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/config/scheduler/JobFactory.java:18]: @Autowired on constructor is only required when we have multiple constructors in our spring components. 
```suggestion
@RequiredArgsConstructor
```

File: src/main/java/uk/gov/hmcts/reform/unspec/config/scheduler/JobScheduler.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+import javax.annotation.PostConstruct;
+
+import static org.quartz.CronScheduleBuilder.cronSchedule;
+import static org.quartz.JobBuilder.newJob;
+import static org.quartz.TriggerBuilder.newTrigger;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/config/scheduler/JobScheduler.java:23]:
```suggestion
@RequiredArgsConstructor
```

File: src/main/java/uk/gov/hmcts/reform/unspec/config/scheduler/SchedulerConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import javax.sql.DataSource;
+
+@Getter
+@Setter
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/config/scheduler/SchedulerConfiguration.java:31]:
```suggestion
@RequiredArgsConstructor
```

File: src/test/java/uk/gov/hmcts/reform/unspec/job/JobSchedulerTest.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.quartz.CronScheduleBuilder.cronSchedule;
+import static org.quartz.TriggerBuilder.newTrigger;
+
+@ExtendWith(SpringExtension.class)
+public class JobSchedulerTest {
```
Comment (sabahirfan) [inline src/test/java/uk/gov/hmcts/reform/unspec/job/JobSchedulerTest.java:34]: `public` can be removed from here

**Review by sabahirfan — DISMISSED** [review]
Few suggestions otherwise LGTM.

**Pull request closed without merging**